subprojects/core/src/main/java/org/gradle/internal/classpath/transforms/AdhocInterceptors.java

```diff
@@ -138,6 +147,20 @@ public boolean visitMethodInsn(MethodVisitorScope mv, String className, int opco
                 return visitINVOKESTATIC(mv, className, owner, name, descriptor);
             case Opcodes.INVOKEVIRTUAL:
                 return visitINVOKEVIRTUAL(mv, className, owner, name, descriptor);
+            case Opcodes.INVOKEINTERFACE:
+                return visitINVOKEINTERFACE(mv, owner, name, descriptor);
+        }
+        return false;
+    }
+
+    private boolean visitINVOKEINTERFACE(MethodVisitorScope mv, String owner, String name, String descriptor) {
+        return interceptGetConvention(mv, owner, name, descriptor);
+    }
+
+    private static boolean interceptGetConvention(MethodVisitorScope mv, String owner, String name, String descriptor) {
```

Comment:
Should this interception go somewhere else?

Reply:
I understand that you followed the same pattern in `visitINVOKEVIRTUAL`. I think it's fine if we inline interceptGetConvention with a comment similar to `// Project.getConvention -> ProjectInterceptors.interceptGetConvention()`